### PR TITLE
[ANCHOR-636] Clean up deprecation for fee_details

### PR DESCRIPTION
**Pull request by JiahuiWho**
This is part of the clean up. 
Replace `amount_fee` and `amount_fee_asset` with `fee_details` to match the txn object

This is a dependency for the major clean up in java-stellar-anchor-sdk to make the E2E tests pass

### CODE REVIEW

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/AnchorTransaction.kt [hunk 1]
```diff
@@ -30,8 +30,7 @@ sealed interface ProcessingAnchorTransaction : AnchorTransaction {
   val amountIn: String
   val amountOutAsset: String?
   val amountOut: String
-  val amountFeeAsset: String?
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/AnchorTransaction.kt:33]: We should keep amountFeeAsset for backward compatibility with older anchors.
We can make amountFee to be optional starting with the new version though

**Review by Ifropc — CHANGES_REQUESTED** [review]

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/AnchorTransaction.kt (hunk 1 above)
Comment (lijamie98) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/AnchorTransaction.kt:33]: The base branch of this PR should be `feature/v3`

File: wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt [hunk 2]
```diff
@@ -46,6 +46,16 @@ data class Payment(
   @SerialName("id_type") val idType: String
 )
 
+@Serializable
+data class FeeDetails(
+  val total: String,
+  val asset: String? = null,
```
Comment (Ifropc) [inline wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt:52]: Asset should be non-null 
Except that LGTM

Comment (JiahuiWho) [reply wallet-sdk/src/main/kotlin/org/stellar/walletsdk/anchor/Data.kt:52]: Hmm just curious why `amountFeeAsset` can be null? I was thinking about matching the nullability

**Review by Ifropc — APPROVED** [review]

**Pull request merged**